realm-annotations-processor/src/test/java/io/realm/processor/RealmProcessorTest.java

```diff
@@ -295,7 +316,7 @@ public void compilePrimaryKeyTypes() throws IOException {
     // Unsupported "PrimaryKey" annotation types
     @Test
     public void compileInvalidPrimaryKeyTypes() throws IOException {
-        final String invalidPrimaryKeyFieldTypes[] = {"boolean", "java.util.Date", "Simple", "RealmList"};
+        final String invalidPrimaryKeyFieldTypes[] = {"boolean", "java.util.Date", "Simple", "RealmList<Simple>", "Boolean"};
```

Comment:
We should allow boolean as an Primary Key IMO. Not doing so is an arbitrary choice since it uses an INTEGER column, and there might be use cases (albeit very small)